projects/vdk-heartbeat/src/vdk/internal/heartbeat/job_controller.py

```diff
@@ -458,3 +457,15 @@ def run(job_input):
             os.path.join(heartbeat_job_dir, "06_override_properties.py"), "w"
         ) as pyfile:
             pyfile.write(python_script)
+
+    @staticmethod
+    def _datetime_from_iso_format(execution: dict):
```

Comment:
From an interface (abstraction) point of view, it would be better the method to accept string and not dictionary (where special key need to be set) , what if we want to use it for start_time . 

Think separation of concerts (https://en.wikipedia.org/wiki/Separation_of_concerns) 
This method deals with parsing dates, while "end_time" is part of the business "layer" and its separate concern.